BusinessCentral.LinterCop.Test/TestCases/Rule0002/NoDiagnostic/1.al

```diff
@@ -0,0 +1,8 @@
+﻿codeunit 50100 MyCodeunit
+{
+    procedure MyProcedure()
+    begin
+        //Comment
+        [|Commit();|]
```

Comment:
I really like your PR.
I don't have experience with this TestFramework.
Can you maybe explain the pattern of the Test AL Objects. 
I don't totally understand this bracket `[]` pattern.

I think that anyone can profit from an Readme on how to ad new tests?

Reply:
The framework is an adaptation of https://github.com/cezarypiatek/RoslynTestKit where I replaced the references to make it compatible with the AL analyzer. The benefit is that you can test minimal examples against single rules without the need to set up a complete AL project.

You can test both positive (a diagnostic should be reported) and negative (no diagnostic should be reported) cases.
There is a test class for each rule that specifies the rule to test and the path to the test files. There can be multiple test files per rule, which are specified in the `[TestCase]` attribute. Currently I just numbered them 1, 2, 3..., because I find it hard to come up with meaningful names.

- TestCases
  - Rule0001
    - HasDiagnostic
      - 1.al
      - ...
    - NoDiagnotic
      - 1.al 
      - ...
  - Rule0002
    - HasDiagnostic
      - ...
    - NoDiagnotic
      - ...
   - ...

The thing with the `[||]` also comes from this framework. You mark the start and end of where the diagnostic should be reported in the code. Theoretically the framework also can work with line numbers, which is less precise, or `TextSpan`, which is more precise, but requires counting characters. I found the `[||]` to be both precise in specifying the location and very easy to read and write.

I will try to include an updated readme until end of the week.